### ISSUE TITLE
Dashboard (replace homepage)

### ISSUE THREAD

**Issue by pavlenex** [btcpayserver/btcpayserver#1711]
**Is your feature request related to a problem? Please describe.**
Currently once you log in into btcpay server, you're greeted with a homepage that's not helpful at all. It's already been discussed that when the time is right, we should replace this homepage with an actual **dashboard**. There are a lot of issues to close before we can get into implementing this one, but I'm opening it for community feedback and ideas on what we can have on a dashboard.

### Homepage

![homepage](https://user-images.githubusercontent.com/36959754/86341360-7c7d0980-bc56-11ea-8238-fe2fcb3fadf5.PNG)

### Describe the solution you'd like
Replace the homepage with a dashboard. Dashboard should provide insights into analytics and important stuff going on on the server. Dashboard should be available for both server admins and regular users, so it should display data from the stores you're owner/member of only. For server admins we can show total number of stores and some analytics there.

### Dashboard example
Dashboard should not infringe the current header navigation of btcpay server. It needs to stay consistent with our design guidelines and provide minimum impact to the rest of the pages.

![dashboard example](https://user-images.githubusercontent.com/36959754/86341767-062cd700-bc57-11ea-9851-5857ff52adc9.PNG)

You can preview the dashboard from above [here](https://preview.pro.ant.design/dashboard/analysis).

One more thing a dashboard should do is perhaps guide user through setting up the store and a wallet properly and provide information on what needs to be done next.

### To-do

- [ ] Determine dashboard design (for designers)
- [ ] Determine important data we want to show
- [ ] Determine the CTA we want to show
- [ ] Determine how we'll show aggregate info for stores

I've been searching for a good example that'll fit into ours for quite a while and the one I outlined above was the best imo. If anybody else has good examples of dashboards, feel free to share them, but be aware they need to provide minimum disruption to the rest of the server.

**Comment by dstrukt** [comment]
I'm working through this and will post some work when I've refined a bit.

**Comment by bolatovumar** [comment]
I agree. I imagine it would be nice for merchants to have a dashboard. Also would be nice to be able to customized it by including different data and being able to add or remove stuff. This doesn't have to be in the first version of course.

**Comment by pavlenex** [comment]
@bolatovumar besides merchant's use-case we can add some data regarding node info, uptime, etc. I think other type of users would like to have such data, but as you said, we can build on it later on. Curious to hear if others have suggestions on what kind of data we can include.

**Comment by Kukks** [comment]
Let's come up with isolatable modules. This can be a great use case for
using view components

**Comment by dstrukt** [comment]
I've been trying to work through some stuff, but I'm not quite ready to share, so I've compiled a few general questions to keep this moving forward, and should hopefully have something to share real soon.

Questions:

1. Time - Is there a global (affects all subsequent views) selector or do want to apply a filter per-component-selector-view?
-- I've thought about this a while and I think it makes more sense as a per-component selector, but I thought I'd keep the question here.

2. Bitcoin Network - Do we address display metrics of the Bitcoin network / price? 
-- Could be as simple as a price ticker and a few basic blocks, or more complex? 
-- I do like the idea of being able to quickly view this dashboard to see if I want/need to dive deeper.
-- This begs the question of what metrics:
- Market Cap, 24H Volume, Circulating Supply, Current Block Height, Chain Size (GB), Hash-rate, etc...

3. Total earnings .. or total held in all wallets? Is this possible? Also open to other alternatives.

4. Do want to show a breakdown of total sales? (e.g. 30% from PoS, 35% from Crowdfund, etc...) Is this even possible?

5. Do we want to show a recent transaction log? This could be a surfacing paid invoices or show your recent pay outs, etc... Open to alternatives. 

6. Do we want to include any additional links, content, etc.. on the bottom of the page below the dashboard views? (I tried to strip out and simplify the bare essentials, but still nice-to-haves)
-- Educational Resources Links/Video?
-- Donate Link
-- Contribute Link


Potential Metrics (feel free to add or modify):
1. Wallets
- Number of (Active) Wallets
- Total # of BTC in all Wallets?

2. Stores 
- Number of (Active) Stores
A. Do we allow the user to manually select each of their stores via dropdown to view following stats?
- Number of Sales
- Conversion Rate (if tracking views or other metric?)
- Number of Views
- Number of Unpaid / Paid Invoices? (or would this fall into a notification .. seems like it might be a nice reminder here)

3. Apps
- Total number of (Active) Apps
A. Do we allow the user to manually select each of their apps via dropdown to view following stats?
- Target Goal
- Amount Raised
- % of Goal
- Number of Contributors

4. Your Node
- Uptime %
- Current State
- Version of Bitcoin

I feel like there's a bit I'm missing here, but to kick the discussion off.

**Comment by bolatovumar** [comment]
> @Kukks Agree 100%
> 
> I've been trying to work through some stuff, but I'm not quite ready to share, so I've compiled a few general questions to keep this moving forward, and should hopefully have something to share real soon.
> 
> Questions:
> 
> 1. Time - Is there a global (affects all subsequent views) selector or do want to apply a filter per-component-selector-view?
>    -- I've thought about this a while and I think it makes more sense as a per-component selector, but I thought I'd keep the question here.

I'm not entirely sure what's meany by "time" here. Are you thinking of allowing filtering over a period of time for some of the stats or metrics? If that's the case then having it be configurable per-component would make sense to me.

> 2. Bitcoin Network - Do we address display metrics of the Bitcoin network / price?
>    -- Could be as simple as a price ticker and a few basic blocks, or more complex?
>    -- I do like the idea of being able to quickly view this dashboard to see if I want/need to dive deeper.
>    -- This begs the question of what metrics:
>    ---- Market Cap, 24H Volume, Circulating Supply, Current Block Height, Chain Size (GB), Hash-rate, etc...

I think we can include anything here really. I feel as from the perspective of most merchants these stats probably don't mean much for their day-to-day operations so which stats are shows won't be critical. If we get feedback about adding X or Y over time, these things can be added.

> 3. Total earnings .. or total held in all wallets? Is this possible? Also open to other alternatives.

"Anything is possible, it's just a matter of time and effort" 
  -- Totally not a made up quote from a totally not a made up wise person.

One thing I would consider here is if it's safe to show wallet amount right away from the privacy perspective.

> 4. Do want to show a breakdown of total sales? (e.g. 30% from PoS, 35% from Crowdfund, etc...) Is this even possible?

I don't see why it wouldn't be possible. I guess this kind of breakdown would only be useful to those users who have more than one active app at all times. If you are a merchant who sells giftcards for Dogecoin, for example, then you probably only have one app which generates most of your revenue so you won't care for a per-app breakdown.

> 5. Do we want to show a recent transaction log? This could be a surfacing paid invoices or show your recent pay outs, etc... Open to alternatives.

This sounds cool.

> 6. Do we want to include any additional links, content, etc.. on the bottom of the page below the dashboard views? (I tried to strip out and simplify the bare essentials, but still nice-to-haves)
>    -- Educational Resources Links/Video?
>    -- Donate Link
>    -- Contribute Link

You mean like a footer?

> Potential Metrics (feel free to add or modify):
> 
> 1. Wallets
> 
> * Number of (Active) Wallets
> * Total # of BTC in all Wallets?
> 
> 1. Stores
> 
> * Number of (Active) Stores
>   A. Do we allow the user to manually select each of their stores via dropdown to view following stats?
> * Number of Sales
> * Conversion Rate (if tracking views or other metric?)
> * Number of Views
> * Number of Unpaid / Paid Invoices? (or would this fall into a notification .. seems like it might be a nice reminder here)
> 
> 1. Apps
> 
> * Total number of (Active) Apps
>   A. Do we allow the user to manually select each of their apps via dropdown to view following stats?
> * Target Goal
> * Amount Raised
> * % of Goal
> * Number of Contributors

I feel like it's ideal to avoid having users select from a dropdown. I'm thinking of having separate card the dashboard for each app. So, if you have 2 POS apps and 1 crowdfund app, you get three cards. You can then configure each card as you see fit (assuming they are configurable, they don't have to be for the first version).
 
> 1. Your Node
> 
> * Uptime %
> * Current State
> * Version of Bitcoin

I feel like this would be a good place to have the "Update BTCPay Server" button as well.

**Comment by NicolasDorier** [comment]
I think for the dashboard it need to be done in "parts" if you take example on Graphana then the dashboard is a grid and components plugged into the dashboard fit into the grid (like 2 colum 1 row for a graph) this allow to optimize the space used in the dashboard.

For example:

![image](https://user-images.githubusercontent.com/3020646/155449134-09bf22bb-ffba-43db-bc1c-7cacfb8cbf8b.png)

Not only this save space, but this would allow for later customization of the dashboad.
I am unsure how easy or difficult it is to do that though. @dennisreimann  is it realistic?

**Comment by dstrukt** [comment]
Agree with the modular flexible design approach, esp. to future proof it as the dashboard grows and we incorporate more cards. I didn't intend for anything to be "fixed" in that respect, what you're seeing in the mock(s) is a 2/3 and 1/3 example to get us rolling and what i felt was the best use of space given the current design requirements for a V1 MVP ... but as we move forward, we could easily do a 1/4, 1/2, 1/3, 2/3, 3/4 or a 4/4 "full-width" within the parent container, as is the case in the Graphana example, afaict. 

It might make more sense to make some things a "1/2" instead of the 2/3, as the 2/3 is taking up too much of the row, but I think if we build with modularity and flexibility in mind, it shouldn't be much of a problem to change down the road.

**Comment by NicolasDorier** [comment]
@dstrukt I pointed it out, because I saw white space. Also I think the unit should be squares. (so Top sales by sales or What's new should be squares)

**Comment by NicolasDorier** [comment]
I don't know how easy it is to do in CSS, if it requires too much ninja css, not worth it to change though @dennisreimann

**Comment by NicolasDorier** [comment]
@dstrukt my bad. It seems some units are indeed rectangle in graphana! Shit how do they do this.

**Comment by NicolasDorier** [comment]
So I checked how they do on https://play.grafana.org/ we can't do that complicated for sure.
So I think fixed grid with number of row/column for each component is good. Just minimizing the whitespace is enough.
